From f7b641bbe85588f84f6e5722533f7ef9e1433d7f Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 31 May 2012 09:56:24 +0200
Subject: [PATCH] dracut-shutdown.service: do not unpack old initramfs on
 shutdown

Do not unpack the initramfs, if the old initramfs still exists.
---
 dracut-shutdown.service |    1 +
 1 file changed, 1 insertion(+)

diff --git a/dracut-shutdown.service b/dracut-shutdown.service
index 4f06e35..6756355 100644
--- a/dracut-shutdown.service
+++ b/dracut-shutdown.service
@@ -11,6 +11,7 @@ After=getty@tty1.service prefdm.service
 Before=reboot.service
 DefaultDependencies=no
 ConditionPathExists=/run/initramfs/.need_shutdown
+ConditionPathExists=!/run/initramfs/bin/sh
 
 [Service]
 ExecStart=/usr/lib/dracut/dracut-initramfs-restore
